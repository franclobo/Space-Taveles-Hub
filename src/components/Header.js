import React from 'react';
import Navbar from './Navbar';

function Header() {
  return (
    <header>
      <img src="../planet.png" alt="space logo" />
      <h1>Space traveler&apos;s Hub</h1>
      <Navbar />
    </header>
  );
}

export default Header;
