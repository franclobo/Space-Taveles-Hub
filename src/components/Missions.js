import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { loadMissions, joinMission, leaveMission } from '../redux/Missions/missions';

// Display Missions.
function Missions() {
  const dispatch = useDispatch();
  const mission = useSelector((state) => state.missions);
  useEffect(() => {
    dispatch(loadMissions());
  }, []);
  const handleJoin = (id) => {
    dispatch(joinMission(id));
  };
  const handleLeave = (id) => {
    dispatch(leaveMission(id));
  };

  return (
    <table className="mission-container">
      <thead>
        <tr className="tableHead">
          <td className="missionTitle"><h1>Mision</h1></td>
          <td className="mission-description"><h1>Description</h1></td>
          <td className="missionStatus"><h1 className="status">Status</h1></td>
          <td />
        </tr>
      </thead>
      <tbody>
        {
          mission.map((item) => (
            <tr className="tableBody" key={item.id}>
              <td className="missionTitle"><h1>{item.name}</h1></td>
              <td className="mission-description"><p>{item.description}</p></td>
              <td className="missionStatus">
                { item.joined ? (
                  <button className="notMember" type="button" onClick={() => handleLeave(item.id)} disabled>Active Member</button>
                ) : (
                  <button className="member" type="button" onClick={() => handleJoin(item.id)} disabled>NOT A MEMBER</button>
                ) }
              </td>
              <td>
                { item.joined ? (
                  <button className="leave" type="button" onClick={() => handleLeave(item.id)}>Leave Mission</button>
                ) : (
                  <button className="join" type="button" onClick={() => handleJoin(item.id)}>Join Mission</button>
                ) }
              </td>
            </tr>
          ))
        }
      </tbody>
    </table>
  );
}

export default Missions;
