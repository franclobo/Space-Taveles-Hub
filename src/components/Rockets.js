import { useEffect, useState } from 'react';
import axios from 'axios';

// fetch rockets
function Rockets() {
  const [data, setData] = useState([]);
  useEffect(() => {
    const fetchData = async () => {
      const res = await axios('https://api.spacexdata.com/v3/rockets');
      setData(res.data);
    };
    fetchData();
  }, []);

  return (
    <div className="rocket-container">
      {
        data.map((item) => (
          <div key={item.rocket_id}>
            <h1>{item.rocket_name}</h1>
            <img src={item.flickr_images[1]} alt="rocket" />
            <p>{item.description}</p>
            <button type="button">Reserve Rocket</button>
          </div>
        ))
      }
    </div>
  );
}

export default Rockets;
