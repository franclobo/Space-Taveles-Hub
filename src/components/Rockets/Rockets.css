.rockets {
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.rocketContainer {
  display: flex;
  gap: 1rem;
  margin: 1rem;
}

.rocketContainer img {
  height: 200px;
  width: 265px;
}

.rocketInfo {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  margin: 1rem;
}

.reserved {
  font-weight: bolder;
  text-align: center;
  color: azure;
  background-color: cadetblue;
  padding: 0.2rem;
  margin-right: 0.4rem;
  border-radius: 5px;
}

.cancelBtn {
  background-color: azure;
  color: darkgrey;
  width: 8rem;
  height: 2rem;
  border-radius: 5px;
  border: 2px slategray solid;
}

.reserveBtn {
  background-color: blue;
  color: azure;
  width: 8rem;
  height: 2rem;
  border-radius: 5px;
  border: none;
}
