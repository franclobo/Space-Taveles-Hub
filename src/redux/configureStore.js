import {
  legacy_createStore as createStore, combineReducers, compose, applyMiddleware,
} from 'redux';
import thunk from 'redux-thunk';

import bookReducer from './books/Books';
import categoriesReducer from './categories/Categories';

const rootReducer = combineReducers({
  books: bookReducer,
  categories: categoriesReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

function generateStore() {
  const store = createStore(rootReducer, composeEnhancers(applyMiddleware(thunk)));
  return store;
}

export default generateStore;
