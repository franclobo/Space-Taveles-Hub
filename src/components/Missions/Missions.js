import { useSelector, useDispatch } from 'react-redux';
import { FetchMission, joinMission, leaveMission } from '../../redux/Missions/missions';
import './Missions.css';

export default function Missions() {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();
  const handleJoin = (id) => {
    dispatch(joinMission(id));
  };

  const handleLeave = (id) => {
    dispatch(leaveMission(id));
  };

  if (missions.length === 0) {
    dispatch(FetchMission());
  }
  return (
    <div>
      <table className="mission-container">
        <thead>
          <tr className="tableHead">
            <td className="missionTitle"><h1>Mision</h1></td>
            <td className="mission-description"><h1>Description</h1></td>
            <td className="missionStatus"><h1 className="status">Status</h1></td>
            <td />
          </tr>
        </thead>
        <tbody>
          {
          missions.map((mission) => (
            <tr className="tableBody" key={mission.id}>
              <td className="missionTitle"><h1>{mission.name}</h1></td>
              <td className="mission-description"><p>{mission.description}</p></td>
              <td className="missionStatus">
                { mission.joined ? (
                  <button className="member" type="button" onClick={() => handleLeave(mission.id)} disabled>Active Member</button>
                ) : (
                  <button className="notMember" type="button" onClick={() => handleJoin(mission.id)} disabled>NOT A MEMBER</button>
                ) }
              </td>
              <td>
                { mission.joined ? (
                  <button className="leave" type="button" onClick={() => handleLeave(mission.id)}>Leave Mission</button>
                ) : (
                  <button className="join" type="button" onClick={() => handleJoin(mission.id)}>Join Mission</button>
                ) }
              </td>
            </tr>
          ))
        }
        </tbody>
      </table>
    </div>
  );
}
