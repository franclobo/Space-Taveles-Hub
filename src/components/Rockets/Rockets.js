import { useSelector, useDispatch } from 'react-redux';
import { FetchRocket } from '../../redux/Rockets/Rockets';
import './Rockets.css';
import Rocket from './rocket';

export default function Rockets() {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  if (rockets.length === 0) {
    dispatch(FetchRocket());
  }

  return (
    <div>
      <ul className="rockets">
        {
          rockets.map((rocket) => (
            <Rocket
              key={rocket.id}
              id={rocket.id}
              name={rocket.name}
              description={rocket.description}
              img={rocket.img}
              reserved={rocket.reserved}
            />
          ))
        }
      </ul>
    </div>
  );
}
