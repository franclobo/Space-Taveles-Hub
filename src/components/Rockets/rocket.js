import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { CancelReserve, ReserveRocket } from '../../redux/Rockets/Rockets';

function Rocket(props) {
  const {
    id, name, description, img, reserved,
  } = props;

  const dispatch = useDispatch();

  const ReserveRocketHandler = () => {
    dispatch(ReserveRocket(id));
  };

  const CancelRocketHandler = () => {
    dispatch(CancelReserve(id));
  };

  return (
    <div className="rocketContainer" id={id}>
      <img src={img} alt="Rocket" />
      <div className="rocketInfo">
        <h4>{name}</h4>
        <div className="description">
          {reserved && (
            <div>
              <p>
                <span className="reserved">Reserved</span>
                {description}
              </p>
            </div>
          )}
          {!reserved && (
            <div>
              <p>{description}</p>
            </div>
          )}
        </div>
        {reserved && (
          <button
            type="button"
            className="cancelBtn"
            onClick={() => {
              CancelRocketHandler();
            }}
          >
            Cancel Reserve
          </button>
        )}
        {!reserved && (
          <button
            type="button"
            className="reserveBtn"
            onClick={() => {
              ReserveRocketHandler();
            }}
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </div>
  );
}

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  reserved: PropTypes.bool,
};

Rocket.defaultProps = { reserved: false };

export default Rocket;
