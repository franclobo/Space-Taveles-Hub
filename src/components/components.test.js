import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from '../redux/configureStore';
import Header from './Header';
import Missions from './Missions';
import Rockets from './Rockets';

test('Header renders correctly', () => {
  const store = configureStore();
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Missions renders correctly', () => {
  const store = configureStore();
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Missions />
      </MemoryRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Rockets renders correctly', () => {
  const store = configureStore();
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Rockets />
      </MemoryRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
