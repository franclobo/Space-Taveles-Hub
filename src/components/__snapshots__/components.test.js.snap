// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Array [
  <header>
    <img
      alt="space logo"
      className="logo"
      src="planet.png"
    />
    <h1
      className="title"
    >
      Space travelers' Hub
    </h1>
    <nav
      className="navBar"
    >
      <a
        aria-current="page"
        className="link active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        className="link"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <a
        className="link"
        href="/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </nav>
  </header>,
  <div
    className="line"
  />,
]
`;

exports[`Missions renders correctly 1`] = `
<table
  className="mission-container"
>
  <thead>
    <tr
      className="tableHead"
    >
      <td
        className="missionTitle"
      >
        <h1>
          Mision
        </h1>
      </td>
      <td
        className="mission-description"
      >
        <h1>
          Description
        </h1>
      </td>
      <td
        className="missionStatus"
      >
        <h1
          className="status"
        >
          Status
        </h1>
      </td>
      <td />
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Rockets renders correctly 1`] = `
<div
  className="rocket-container"
/>
`;
